dev/run-tests-jenkins.py

```diff
@@ -0,0 +1,256 @@
+
+        scp_proc = subprocess.Popen(['scp', log_archive,
+                                     'amp-jenkins-master:' + jenkins_build_dir + '/' + log_archive],
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+        scp_stdout, scp_stderr = scp_proc.communicate()
+        scp_returncode = scp_proc.returncode
+
+        if not scp_returncode == 0:
+            print_err("Failed to send archived unit tests logs to Jenkins master.")
+            print_err(" > scp_status: %s" % scp_returncode)
+            print_err(" > scp_output: %s" % scp_stdout)
+        else:
+            print(" > Send successful.")
+    else:
+        print_err(" > No log files found.")
+
+        rm_r(log_archive)
+
+
+def run_pr_checks(pr_tests, ghprb_actual_commit, sha1):
+    """
+    Executes a set of pull request checks to ease development and report issues with various
+    components such as style, linting, dependencies, compatibilities, etc.
+    @return a list of messages to post back to Github
+    """
+    # Ensure we save off the current HEAD to revert to
+    current_pr_head = run_cmd(['git', 'rev-parse', 'HEAD'], return_output=True).strip()
+    pr_results = list()
+
+    for pr_test in pr_tests:
+        pr_results.append(run_cmd(['bash', os.path.join(SPARK_HOME, 'dev', 'tests', pr_test),
+                                   ghprb_actual_commit, sha1],
+                                  return_output=True).strip())
+        # Ensure, after each test, that we're back on the current PR
+        run_cmd(['git', 'checkout', '-f', current_pr_head])
+    return pr_results
+
+
+def pr_message(build_display_name,
+               build_url,
+               ghprb_pull_id,
+               short_commit_hash,
+               commit_url,
+               msg,
+               post_msg=''):
+    # align the arguments properly for string formatting
+    str_args = (build_display_name,
+                msg,
+                build_url,
+                ghprb_pull_id,
+                short_commit_hash,
+                commit_url,
+                str(' ' + post_msg + '.') if post_msg else '.')
+    return '**[Test build %s %s](%sconsole)** for PR %s at commit [\`%s\`](%s)%s' % str_args
+
+
+def run_tests(tests_timeout):
+    """
+    Runs the `dev/run-tests` script and responds with the correct error message
+    under the various failure scenarios.
+    @return a tuple containing the test result code and the result note to post to Github
+    """
+
+    test_proc = subprocess.Popen(['timeout',
+                                  tests_timeout,
+                                  os.path.join(SPARK_HOME, 'dev', 'run-tests')]).wait()
+    test_result_code = test_proc.returncode
+
+    def failure_result_note(msg):
+        ' * This patch **fails ' + msg + '**.'
+
+    failure_note_by_errcode = {
+        ERROR_CODES["BLOCK_GENERAL"]: failure_result_note('some tests'),
```

Comment:
I was a bit inprecise in my earlier comment, but I was thinking that we could change this so that `failure_note_by_errcode` holds the fragments, e.g.

```python
ERROR_CODES["BLOCK_GENERAL"]: 'some tests',
```

and we could then index into this array with the test result error code, e.g.

```python
' * This patch **fails %s **.' % failure_note_by_errcode[test_result]
```

which I think is slightly clearer.